### PR TITLE
UIU-2969: change user type confirmation  modal

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[https://folio-org.atlassian.net/browse/UIU-2969](https://folio-org.atlassian.net/browse/UIU-2969)
## Approach

https://github.com/folio-org/ui-users/assets/116072773/8e4bf2a1-cf4d-4127-b271-5e2e31ddebff

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/EditSections/EditUserInfo/EditUserInfo.test.js [hunk 1]
```diff
@@ -195,6 +209,31 @@ describe('Render Edit User Information component', () => {
...
 
+  it('should display change user type modal', async () => {
+    renderEditUserInfo({
+      ...props,
+      initialValues: {
+        ...props.initialValues,
+        type: USER_TYPES.STAFF,
+      },
+    });
+    // expect(screen.getByText('ChangeUserTypeModal')).toBeInTheDocument();
```
Comment (usavkov-epam) [inline src/components/EditSections/EditUserInfo/EditUserInfo.test.js:220]: comment

File: src/components/EditSections/EditUserInfo/components/ChangeUserTypeModal/ChangeUserTypeModal.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  return (
+    <Modal
+      footer={userTypeModalFooter}
+      id="userType_confirmation_modal"
+      label={<FormattedMessage id="ui-users.information.change.userType.modal.label" />}
+      open={open}
+    >
+      <div>
+        <FormattedMessage
+          id="ui-users.information.change.userType.modal.text"
```
Comment (usavkov-epam) [inline src/components/EditSections/EditUserInfo/components/ChangeUserTypeModal/ChangeUserTypeModal.js:39]: it makes sense to keep one row for a component with one prop

**Review by usavkov-epam — APPROVED** [review]

File: translations/ui-users/en.json [hunk 3]
```diff
@@ -335,6 +335,9 @@
   "information.recalculate.modal.button": "Set",
   "information.recalculate.modal.text": "Library accounts with patron group <b>{group}</b> expire in <b>{offset} days</b>. Do you want to set this user’s account to expire on <b>{date}</b>?",
   "information.recalculate.modal.label": "Set expiration date?",
+  "information.change.userType.modal.label": "Changing user type?",
+  "information.change.userType.modal.button": "Set",
```
Comment (Terala-Priyanka) [inline translations/ui-users/en.json:339]: As per the requirement stated in ticket, shouldn't this be "Confirm"?

File: src/components/EditSections/EditUserInfo/EditUserInfo.test.js (hunk 1 above)
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/EditUserInfo.test.js:220]: remove the comment

File: src/components/EditSections/EditUserInfo/EditUserInfo.test.js [hunk 4]
```diff
@@ -195,6 +209,31 @@ describe('Render Edit User Information component', () => {
...
+
+
+    await userEvent.click(option);
+
+    await screen.findByText('ChangeUserTypeModal');
+
+    const cancelButton = screen.getByText('Cancel');
+
+    await userEvent.click(cancelButton);
+    expect(changeMock).toHaveBeenCalled();
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditUserInfo/EditUserInfo.test.js:234]: The title of the test and assertion in expect do not match.

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]
The Confirmation modal is being displayed on non-ECS environment too.
As per the ticket, the functionality applies to ECS enabled environment.
Please incorporate an ECS check.

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,6 +23,7 @@
 * Enable effective call number column sorting in Open Loans screen. Refs UIU-3002.
 * User Information in User Edit to display profile picture and update button set. Refs UIU-3005.
 * Update request header for pay several Fees/fines. Refs UIU-3040.
+* Changing user type confirmation modal. Refs UIU-2969.
```
Comment (Terala-Priyanka) [inline CHANGELOG.md:26]: Please update the description to state that it is for ECS enabled env.

Comment (alisher-epam) [reply CHANGELOG.md]: done

**Review by Terala-Priyanka — APPROVED** [review]

**Comment by Terala-Priyanka** [comment]
@alisher-epam 
looks like there is a new issue reported by Sonar. Please fix it.

**Comment by Terala-Priyanka** [comment]
@alisher-epam 
looks like the issue injected by your changes persists. 
It was expected and agreed by you that it would be fixed.
Please fix it ASAP.

**Comment by NikitaSedyx** [comment]
@Terala-Priyanka render function already has high complexity, mb it's time for you to do a refactoring?

his condition is reasonable in `render` as it's part of `OnChange` (it could be improved, but still ok), when functions that are created inside `render` and contain conditions are definitely not  expected (I have no idea who introduced them), so instead of fixing normal code just to fix, may be bad code should be refactored so sonar doesn't complain?

**Comment by Terala-Priyanka** [comment]
Just an FYI..
another PR https://github.com/folio-org/ui-users/pull/2636/ with master as target, doesn't have this sonar complaint. 
I did not dig much into the details

**Comment by NikitaSedyx** [comment]
just because someone could remove one of the conditions, but the point is the same, to ask to sustain quality - it should be initially in good state (I'm talking only for cases like this one, and not if quality is not so good we can do whatever we want)

**Pull request merged**